### PR TITLE
Migrate protobuf logic to plugin

### PR DESCRIPTION
**Pull request by anuraaga**
Also remove IDE logic which seems to have been fixed in the protobuf plugin at some point

### CODE REVIEW

**Review by kubawach — APPROVED** [review]
Looks good!

File: build.gradle.kts [hunk 1]
```diff
@@ -107,37 +104,6 @@ subprojects {
...
-            afterEvaluate {
-                // Classpath when compiling protos, we add dependency management directly
-                // since it doesn't follow Gradle conventions of naming / properties.
-                dependencies {
-                    add("compileProtoPath", platform(project(":dependencyManagement")))
-                    add("testCompileProtoPath", platform(project(":dependencyManagement")))
-                }
-            }
-        }
-
```
Comment (jkwatson) [inline build.gradle.kts:140]: whoa. this stuff isn't needed any more?

Comment (jkwatson) [reply build.gradle.kts:140]: oh wait...it's just moved to the other file. 🤦🏼

**Review by jkwatson — APPROVED** [review]
By the way, if you just do a project "build" in IDEA, does it work for you? For me, it partially fails to due some of the jmh dependencies not being resolved, even though IDEA doesn't show any errors in the files.

**Pull request merged**